test/repository_test.rb

```diff
@@ -115,6 +120,20 @@
           UserRepository.create(user1)
           user1.id.must_equal id
         end
+
+        describe 'when entity is already persisted' do
+          it 'assigns and persists created_at attribute' do
+            result = UserRepository.create(@unpersisted_user)
```

Comment:
This is wrong, for  unpersisted entity, you don't need to use call repository, just instanstiate the entity